framework/packages/abstract-client/tests/integration.rs

```diff
@@ -813,3 +813,21 @@ fn can_get_abstract_account_from_client_account() -> anyhow::Result<()> {
     assert_eq!(abstract_account.id()?, AccountId::local(1));
     Ok(())
 }
+
+#[test]
+fn can_customize_sub_account() -> anyhow::Result<()> {
+    let client = AbstractClient::builder(Mock::new(&Addr::unchecked(OWNER))).build()?;
+    let account = client.account_builder().build()?;
+    let sub_account = client
+        .account_builder()
+        .name("foo-bar")
+        .ownership(GovernanceDetails::SubAccount {
+            manager: account.manager()?.to_string(),
+            proxy: account.proxy()?.to_string(),
+        })
+        .build()?;
```

Comment:
@adairrr Would you consider it good enough way and I should just add documentation on how to do it? Or would you prefer to have customization when installing app from account itself, instead of pre-defined fields?

Reply:
can it have `fn sub_account(&mut self, owner_account: &Account) -> &mut Self`?

Kinda weird you need to provide sub-account governance details yourself.

Reply:
And IMO the build should throw an err if ownership is set to sub_account is this is an action that needs to be called on the owning account, so attempting to create one manually should fail.

Reply:
> can it have `fn sub_account(&mut self, owner_account: &Account) -> &mut Self`?
> 
> Kinda weird you need to provide sub-account governance details yourself.

Should we hide `ownership()` method then?

Reply:
Ah, interesting. This isn't actually enforced in our code... This means that anyone can create sub-accounts on other people's accounts I think.

Reply:
> > can it have `fn sub_account(&mut self, owner_account: &Account) -> &mut Self`?
> > Kinda weird you need to provide sub-account governance details yourself.
> 
> Should we hide `ownership()` method then?

I don't think so. I don't see anyone attempting to create sub-accounts this way, but should be commented that it's not possible.